sklearn/preprocessing/_encoders.py

```diff
@@ -57,8 +87,12 @@ def _check_X(self, X):
 
         for i in range(n_features):
             Xi = self._get_feature(X, feature_idx=i)
-            Xi = check_array(Xi, ensure_2d=False, dtype=None,
-                             force_all_finite=needs_validation)
+            if self.categories == 'dtypes' and hasattr(Xi, 'cat'):
```

Comment:
The hasattr check could also be `Xi.dtype.name == 'category'`, not sure which one is better